server/controllers/metricsController.js

```diff
@@ -1,35 +1,75 @@
 const fetch = require('node-fetch');
+const cmd = require('node-cmd');
+const { spawn } = require('child_process');
 
 const metricsController = {};
 
 const PROM_URL = 'http://127.0.0.1:9090/api/v1/';
 
+let isPromUp = false;
+const forwardPromPort = () =>
+  new Promise((resolve, reject) => {
+    const promPodName = cmd
+      .runSync(
+        'export POD_NAME=$(kubectl get pods --all-namespaces -l "app=prometheus,component=server" -o jsonpath="{.items[0].metadata.name}") && echo $POD_NAME',
+      )
+      .data
+      .split('\n');
+      
+    const portForward = spawn('kubectl', [
+      '--namespace=prometheus',
+      'port-forward',
+      `${promPodName[0]}`,
+      '9090',
+    ]);
+
+    portForward.stdout.on('data', (data) => {
+      console.log(`stdout: ${data}`);
+      isPromUp = true;
+      resolve();
+    });
+
+    portForward.stderr.on('data', (data) => {
+      console.log(`stderr: ${data}`);
+    });
+
+    portForward.on('close', (code) => {
+      if (code === 1) console.log('Prometheus is already running...');
```

Comment:
should we still hit lines 38 and 39 if 37 is true?

Reply:
yup, those statement run in case Prometheus is already running in the background prior to the program starting. It will then show that prometheus is running by updating the isPromUp variable.